cnpy.cpp

```diff
@@ -63,8 +63,13 @@ void cnpy::parse_npy_header(unsigned char* buffer,size_t& word_size, std::vector
     //std::string magic_string(buffer,6);
     uint8_t major_version = *reinterpret_cast<uint8_t*>(buffer+6);
     uint8_t minor_version = *reinterpret_cast<uint8_t*>(buffer+7);
-    uint16_t header_len = *reinterpret_cast<uint16_t*>(buffer+8);
-    std::string header(reinterpret_cast<char*>(buffer+9),header_len);
+    bool extended_header = (major_version > 1);
+    uint32_t header_len;
+    if (extended_header)
+      header_len = *reinterpret_cast<uint32_t*>(buffer+8);
+    else
+      header_len = *reinterpret_cast<uint16_t*>(buffer+8);
+    std::string header(reinterpret_cast<char*>(buffer+(extended_header ? 11 : 9)),header_len);
```

Comment:
Since `header_len` is `uint16_t`, I think it should be?

```suggestion
    std::string header(reinterpret_cast<char*>(buffer+(extended_header ? 12 : 10)),header_len);
```

Reply:
I think you're right. It's been working because the only use of `header` has been based on relative locations discovered using `find` and similar, but we could improve performance (ever so slightly) by using the correct offset.

Reply:
Fixed. Thanks for pointing this out.